### ISSUE TITLE
Basecamp Integration

### ISSUE THREAD

**Issue by ajtroxell** [gogits/gogs#1653]
Has there been any talk of Basecamp integration like what is available with Beanstalk?

**Comment by Unknwon** [comment]
Hi @ajtroxell !

I haven't had experience with Basecamp or Beanstalk, so...

What kinds of integrations do you want to have?

**Comment by ajtroxell** [comment]
The possibility to tie into the Basecamp API (https://github.com/basecamp/api) in order to mark to-dos as completed much like is possible with issues being mentioned within the commit message. Perhaps with some kind of built in post-receive functionality that picks up the todo number that is referenced in the commit like <code>todo #204954163<code>. Documentation on the api and marking a todo as completed is outlined here: https://github.com/basecamp/basecamp-classic-api/blob/master/sections/todo_list_items.md

Or perhaps due to the client review process it instead posts the commit message to the todo instead of closing it as this would give time for a client or other party to review the work done on the todo before it is marked as completed.

It could be taken many different directions though. Commit made that references todo into staging branch, posts a comment with the commit message to the todo. If that same commit message is seen on the master branch the issue is then closed, but i fear that may get too complicated.

**Comment by Unknwon** [comment]
Thanks for following up and detail.

In case you're not aware, Gogs has supported Git hooks with permission must be given by admin, not sure if it satisfies your needs.

**Comment by ajtroxell** [comment]
We utilize hooks now for deployments to specific environments. I was only thinking a more UI based approach would work better for configuring something like this. Mostly due to the popularity of Basecamp, and the very large benefit this would give Gogs over other options.

**Comment by Unknwon** [comment]
@ajtroxell full integration would be awesome as always, and obviously it takes huge time and hangs all the other critical tasks as always.

But I can implement a webhook-based integration first, is it sounds interesting to you?

**Comment by ajtroxell** [comment]
This would be very interesting. Perhaps we could fork and take the reigns and do a more thorough integration via the UI after that if we can find the time.

**Comment by Unknwon** [comment]
Thanks!

Does Basecamp has a general purpose webhook thing like Slack or other similar services?

**Comment by ChangePlaces** [comment]
If there's too tight an integration, what about users who have absolutely no interest in basecamp at all? I'd hate to see gogs become too tied down with other software

**Comment by tboerger** [comment]
@ChangePlaces what do you mean? This is just an optional integration that you can get notified within basecamp about new issues, pull requests or pushs. That's nothing that slows down or influencing bad the gogs behavior.

**Comment by ajtroxell** [comment]
Yes it would be a a optional integration that would simply "ping" the Basecamp API to create replies to existing to-do tasks.





On Tue, Oct 27, 2015 at 1:35 PM -0700, "ChangePlaces" <notifications@github.com> wrote:
If there's too tight an integration, what about users who have absolutely no interest in basecamp at all? I'd hate to see gogs become too tied down with other software

---

**Comment by ChangePlaces** [comment]
ah my misunderstanding then!
